packages/api/tests/lambdas/test-sf-event-sqs-to-db-records.js

```diff
@@ -287,194 +314,162 @@ test.serial('hasNoAsyncOpOrExists returns false if async operation does not exis
+          };
+        }
+        return undefined;
+      },
+    }),
+  });
 
-  t.false(await hasNoCollectionOrExists({
-    meta: {
-      collection,
-    },
-  }, knex));
-  t.true(doesRecordExistStub.called);
+  t.deepEqual(
+    await getMessageProvider(cumulusMessage, fakeKnex),
+    {
+      cumulusId: 234,
+    }
+  );
 });
 
-test('shouldWriteExecutionToRDS returns false for pre-RDS deployment execution message', async (t) => {
+test('getMessageProvider returns undefined if provider cannot be found', async (t) => {
   const { knex } = t.context;
-  t.false(await shouldWriteExecutionToRDS({
-    cumulus_meta: {
-      cumulus_version: '2.99.1',
+  t.is(await getMessageProvider({}, knex), undefined);
+});
+
+test('shouldWriteExecutionToRDS returns false for pre-RDS deployment execution message', async (t) => {
+  const { cumulusMessage, knex, preRDSDeploymentVersion } = t.context;
+  t.false(await shouldWriteExecutionToRDS(
+    {
+      ...cumulusMessage,
+      cumulus_meta: {
+        ...cumulusMessage.cumulus_meta,
+        cumulus_version: preRDSDeploymentVersion,
+      },
     },
-  }, knex));
+    { cumulusId: 1 },
+    knex
+  ));
 });
 
 test.serial('shouldWriteExecutionToRDS returns true for post-RDS deployment execution message if all referenced objects exist', async (t) => {
   const {
     knex,
     doesRecordExistStub,
+    cumulusMessage,
+    asyncOperationId,
+    parentExecutionArn,
   } = t.context;
 
-  const parentExecutionArn = `machine:${cryptoRandomString({ length: 5 })}`;
-  const asyncOperationId = uuidv4();
-  const collection = {
-    name: 'fake-name',
-    version: '0.0.0',
-  };
-
   doesRecordExistStub.withArgs({
     id: asyncOperationId,
   }).resolves(true);
-  doesRecordExistStub.withArgs(collection).resolves(true);
   doesRecordExistStub.withArgs({
     arn: parentExecutionArn,
   }).resolves(true);
 
   t.true(
-    await shouldWriteExecutionToRDS({
-      cumulus_meta: {
-        cumulus_version: '3.0.0',
-        parentExecutionArn,
-        asyncOperationId,
-      },
-      meta: {
-        collection,
-      },
-    }, knex)
+    await shouldWriteExecutionToRDS(
+      cumulusMessage,
+      { cumulusId: 1 },
+      knex
+    )
   );
 });
 
 test.serial('shouldWriteExecutionToRDS returns false if error is thrown', async (t) => {
   const {
     knex,
     doesRecordExistStub,
+    cumulusMessage,
+    parentExecutionArn,
+    collectionCumulusId,
   } = t.context;
 
-  const parentExecutionArn = `machine:${cryptoRandomString({ length: 5 })}`;
-
   doesRecordExistStub.withArgs({
     arn: parentExecutionArn,
   }).throws();
 
   t.false(
-    await shouldWriteExecutionToRDS({
-      cumulus_meta: {
-        cumulus_version: '3.0.0',
-        parentExecutionArn,
-      },
-    }, knex)
+    await shouldWriteExecutionToRDS(cumulusMessage, { cumulusId: collectionCumulusId }, knex)
+  );
+});
+
+test('shouldWriteExecutionToRDS returns false if collection record is not defined', async (t) => {
```

Comment:
Hmm, this makes me wonder if isPostRDSDeploymentExecution should be throwing at all in this case. 

Is there a reason we shouldn't raise a warning message not throw in that method, so that another throw elsewhere doesn't cause the lambda to halt?